### ISSUE TITLE
RNN output dimension considering hidden units

### ISSUE THREAD

**Issue by xyzxinyizhang** [keras-team/keras#10800]
Hi, 

I have a question for the RNN output dimension.

I use (I simply the model here)
model.add(SimpleRNN(2,input_shape=(None, 1),return_sequences=True,name="1")) 
model.add(SimpleRNN(4,return_sequences=False, name="2"))

Each of my input sequence length is 9 so I got the layer"1" output shape as (2297, 9, 2), 2297 is my total samples, 2 is my output dimension. 
The model works well. But I'm wondering why the RNN output dimension is related to hidden units?
Refer to S_t=(U*X_t+W*S_t), O_t=Softmax(V*St)

For layer 1:
U(hidden*input_dim) is 2*1 matrix, X_t is 1*1 matrix, W(hidden*hidden) is 2*2 matrix, V(input_dim*hidden)=1*2.
Then the O_t should be a 1*1 here. Why my output dimension is 2?

Below is my input example:
[[4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]
 [4.00298875]]

Can anyone help this?
